pkg/controller/metrics/adapters/metrics_sender_to_kafka.go

```diff
@@ -0,0 +1,71 @@
+package adapters
+
+import (
+	"context"
+	log "github.com/golang/glog"
+	"github.com/opensds/opensds/pkg/model"
+	"github.com/segmentio/kafka-go"
+	"strconv"
+)
+
+type KafkaMetricsSender struct {
+	Queue    chan *model.MetricSpec
+	QuitChan chan bool
+}
+
+func (p *KafkaMetricsSender) GetMetricsSender() MetricsSenderIntf {
+	sender := KafkaMetricsSender{}
+	sender.Queue = make(chan *model.MetricSpec)
+	sender.QuitChan = make(chan bool)
+	return &sender
+}
+
+func (p *KafkaMetricsSender) Start() {
+	go func() {
+		for {
+			select {
+			case work := <-p.Queue:
+				// Receive a work request.
+				log.Infof("GetMetricsSenderToKafka received metrics for instance %s\n and metrics %f\n", work.InstanceID, work.Value)
+
+				// do the actual sending work here
+				// make a writer that produces to topic-A, using the least-bytes distribution
+				w := kafka.NewWriter(kafka.WriterConfig{
+					Brokers:  []string{"localhost:9092"},
```

Comment:
Make the port configurable.  At least add a TODO.

Reply:
https://github.com/opensds/opensds/issues/691